Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.mm

```diff
@@ -570,9 +570,17 @@ static void encodeSecureCodingInternal(Encoder& encoder, id <NSObject, NSSecureC
     }
 
 #if ENABLE(REVEAL)
-    // FIXME: This can be removed on operating systems that have rdar://109237983.
+    // FIXME: This can be removed for RVItem on operating systems that have rdar://109237983.
     if (PAL::isRevealCoreFrameworkAvailable() && [object isKindOfClass:PAL::getRVItemClass()])
         [delegate setTransformURLs:NO];
+#endif
+#if ENABLE(DATA_DETECTION)
+    if (PAL::isDataDetectorsCoreFrameworkAvailable() && [object isKindOfClass:PAL::getDDScannerResultClass()])
+        [delegate setTransformURLs:NO];
+#if PLATFORM(MAC)
+    if (PAL::isDataDetectorsFrameworkAvailable() && [object isKindOfClass:PAL::getWKDDActionContextClass()])
+        [delegate setTransformURLs:NO];
+#endif
 #endif
```

Comment:
Nit: This `#endif` could use a comment about what it matches.  (Not necessary to land the PR.)